File: node/getRequest.js
const http = require('http');
const company = "YOUR_TEAMWORK_SITE_NAME";
const key = "YOUR_API_KEY";

const base64 = Buffer.from(`${key}:xxx`).toString("base64");

const options = {
    hostname: `${company}.teamwork.com`,
    path: "/projects.json",
    method: "GET",
    headers: {
        "Authorization": `BASIC ${base64}`,
        "Content-Type": "application/json"
    }
};

const req = http.request(options, function(res) {
    console.log(`STATUS: ${res.statusCode}`);
    console.log(`HEADERS: ${JSON.stringify(res.headers)}`);
    res.setEncoding("utf8");
    res.on("data", function (chunk) {
        console.log(`BODY: ${chunk}`);
    });
});

req.on("error", function(e) {
    console.log(`ERROR: ${error.message}`);
});

req.end();

File: python3/getRequest.py
import urllib3
import certifi

http = urllib3.PoolManager(cert_reqs='CERT_REQUIRED', ca_certs=certifi.where())

company = "YOUR_TEAMWORK_SITE_NAME"
key = "YOUR_API_KEY"
action = "projects.json"

url = "https://{0}.teamwork.com/{1}".format(company, action)
headers = urllib3.util.make_headers(basic_auth=key + ":xxx")
request = http.request('GET', url, headers=headers)

response = request.status
data = request.data

print(data)
